### ISSUE TITLE
[Darwin] Internal objc methods used for unit tests should be declared in separate header

### ISSUE THREAD

**Issue by jtung-apple** [project-chip/connectedhomeip#31966]
Per @bzbarsky-apple 's comments: https://github.com/project-chip/connectedhomeip/pull/31952#discussion_r1480341626

**Comment by bzbarsky-apple** [comment]
No, the idea was for this header to be a project header in the framework, not in the tests, so then the framework itself also includes that header an we ensure the selectors don't diverge, etc.
